acbuild/acbuild.go

```diff
@@ -156,6 +163,41 @@ func stdout(format string, a ...interface{}) {
 	fmt.Fprintln(os.Stdout, strings.TrimSuffix(out, "\n"))
 }
 
+func getLock() error {
+	err := os.MkdirAll(path.Join(contextpath, workprefix), 0755)
+	if err != nil {
+		return err
+	}
+
+	lockfile, err = os.OpenFile(lockpath(), os.O_CREATE|os.O_RDWR, 0644)
+	if err != nil {
+		return err
+	}
+
+	err = syscall.Flock(int(lockfile.Fd()), syscall.LOCK_EX)
```

Comment:
I dunno, I'd consider any situation where there are concurrent invocations
to be a bug (in whatever is invoking acbuild) and for it to fail fast.
Otherwise it seems inherently racy; for the most part acbuild operations
are necessarily sequential. Can you think of a use case I'm missing?

On Wed, Oct 7, 2015 at 12:23 PM, Derek Gonyeo <notifications@github.com>
wrote:

> In acbuild/acbuild.go
> <https://github.com/appc/acbuild/pull/26#discussion_r41434004>:
>
> > @@ -156,6 +163,41 @@ func stdout(format string, a ...interface{}) {
> >  	fmt.Fprintln(os.Stdout, strings.TrimSuffix(out, "\n"))
> >  }
> >
> > +func getLock() error {
> > +	err := os.MkdirAll(path.Join(contextpath, workprefix), 0755)
> > +	if err != nil {
> > +		return err
> > +	}
> > +
> > +	lockfile, err = os.OpenFile(lockpath(), os.O_CREATE|os.O_RDWR, 0644)
> > +	if err != nil {
> > +		return err
> > +	}
> > +
> > +	err = syscall.Flock(int(lockfile.Fd()), syscall.LOCK_EX)
>
> Why not just wait for the lock to be released?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/appc/acbuild/pull/26/files#r41434004>.
>